src/datasets/datasets.service.ts

```diff
@@ -50,8 +73,185 @@ export class DatasetsService {
     }
   }
 
+  getRelationViewAccess(field: DatasetLookupKeysEnum, user: JWTUser) {
+    switch (field) {
+      case DatasetLookupKeysEnum.proposals: {
+        const ability = this.caslAbilityFactory.proposalsInstanceAccess(user);
+        const canViewAny = ability.can(Action.ProposalsReadAny, ProposalClass);
+        const canViewAccess = ability.can(
+          Action.ProposalsReadManyAccess,
+          ProposalClass,
+        );
+        const canViewOwner = ability.can(
+          Action.ProposalsReadManyOwner,
+          ProposalClass,
+        );
+        const canViewPublic = ability.can(
+          Action.ProposalsReadManyPublic,
+          ProposalClass,
+        );
+
+        return { canViewAny, canViewOwner, canViewAccess, canViewPublic };
+      }
+      case DatasetLookupKeysEnum.origdatablocks: {
+        const ability =
+          this.caslAbilityFactory.origDatablockInstanceAccess(user);
```

Comment:
I feel like we are putting too many authorization logic codes into the service file, which supposed to store mainly for database interaction code.
Wouldn't it better to have a `datasets.access.service.ts` file or something similar to separate the logics?